proxy/src/avatar.rs

```diff
@@ -116,34 +123,35 @@ impl Color {
 
     /// Compute the lightness of a color.
     #[must_use]
-    pub fn lightness(&self) -> f32 {
-        let r = self.r as f32;
-        let g = self.g as f32;
-        let b = self.b as f32;
-        let n = 0xff as f32;
+    pub fn lightness(self) -> f32 {
+        let r = f32::from(self.r);
+        let g = f32::from(self.g);
+        let b = f32::from(self.b);
+        let n = 255_f32;
 
         // This isn't perceptual lightness, but whatever.
         (r / n + g / n + b / n) / 3.
     }
 
     /// Ligthen a color by an amount between `-1.0` and `1.0`.
+    #[allow(clippy::shadow_reuse)]
```

Comment:
This lint doesn't seem useful -- shadowing is pretty idiomatic in rust as far as I've seen.

Reply:
Promoted to crate `allow` in 38adaab.